sympy/polys/factortools.py

```diff
@@ -124,13 +124,37 @@ def dmp_trial_division(f, factors, u, K):
 
 
 def dup_zz_mignotte_bound(f, K):
-    """Mignotte bound for univariate polynomials in `K[x]`. """
-    a = dup_max_norm(f, K)
-    b = abs(dup_LC(f, K))
-    n = dup_degree(f)
+    """
+    The Knuth-Cohen variant of Mignotte bound for
+    univariate polynomials in `K[x]`.
 
-    return K.sqrt(K(n + 1))*2**n*a*b
+    References
+    ==========
+
+    ..[1] [Abbott2013]_
+
+    """
+    from sympy import binomial
+
+    d = dup_degree(f)
+    delta = _ceil( d / 2 )
+
+    # euclidean-norm
+    eucl_norm = K.sqrt( sum( [cf**2 for cf in f] ) )
+
+    # biggest values of binomial coefficients (p. 538 of reference)
+    t1 = binomial( delta - 1, _ceil( delta / 2 ) )
```

Comment:
I would consider defining something like `delta2 = _ceil(delta/2)` in addition to `delta` and using it on this and the next line.